editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardRuntimeAndCamelPage.java

```diff
@@ -421,12 +422,21 @@ private void preselectCamelVersionForRuntime(String runtimeCamelVersion) {
 		if (UNKNOWN_CAMEL_VERSION.equals(runtimeCamelVersion)) {
 			camelVersionCombo.setEnabled(true);
 		} else {
+			ArrayList<String> compatibleVersions = new ArrayList<>();
 			for (String camelVersion : camelVersionCombo.getItems()) {
 				if (isCompatible(runtimeCamelVersion, camelVersion)) {
-					camelVersionCombo.setText(camelVersion);
-					return;
+					compatibleVersions.add(camelVersion);
 				}
 			}
+			if (compatibleVersions.size()>0) {
+				Collections.sort(compatibleVersions);
+				if (compatibleVersions.contains(runtimeCamelVersion)) {
+					camelVersionCombo.setText(runtimeCamelVersion);
+				} else {
+					camelVersionCombo.setText(compatibleVersions.get(compatibleVersions.size()-1));
+				}
+				return;
```

Comment:
instead of using an enclosed return, I think that it is mre readbale to use a full if/else:

			if (!compatibleVersions.isEmpty()) {
				Collections.sort(compatibleVersions);
				if (compatibleVersions.contains(runtimeCamelVersion)) {
					camelVersionCombo.setText(runtimeCamelVersion);
				} else {
					camelVersionCombo.setText(compatibleVersions.get(compatibleVersions.size()-1));
				}
			} else {
				camelVersionCombo.select(Math.max(0, camelVersionCombo.getItemCount()-1));
			}